data/util/NumericShaperUtility.js

```diff
@@ -0,0 +1,294 @@
+					//		Converts the digits in the text to European or Arabic digits
+					//		According to the shaperType & the textDir.
+					// description:
+					//      This function is intended to convert the digits in the input 
+					//		Text from European to Arabic & vice versa according to 
+					//		The shaperType & the textDir as the following:
+					//		1-Arabic: if shaperType = 'National'.
+					//      2-Arabic: if shaperType = 'Contextual' & the preceding character is Arabic.
+					//		3-Arabic: if shaperType = 'Contextual' & textDir='rtl' & no preceding strong character.
+					//		4-European: if shaperType = 'Nominal'.
+					//		5-European: if shaperType = 'Contextual' & the preceding character is English.
+					//		6-European: if shaperType = 'Contextual' & textDir='ltr' & no preceding strong character.
+					// text: String
+					//		The text to be shaped.
+					// shaperType: String
+					//		The type of the shaper to be used.
+					//		Allowed values:
+					//		1."National"
+					//		2."Nominal"
+					//		3."Contextual"
+					// textDir: String
+					//		The direction of the input text.
+					//		Allowed values:
+					//		1. "ltr"
+					//		2. "rtl"
+					//		3. "auto"
+					// returns:
+					//		The shaped string.
+					
+					if(!text){
+						return;
+					}
+					
+					if(["National", "Nominal", "Contextual"].indexOf(shaperType) === -1){
+						return text;
+					}
+					
+					if(shaperType === "National"){
+						return this._shapeArabic(text);
+					}else if(shaperType === "Nominal"){
+						return this._shapeEuropean(text);
+					}else if(shaperType === "Contextual" && textDir === "rtl"){
+						this._context = 2;
+						return this._shapeContextual(text);
+					}else{
+						this._context = 1;
+						return this._shapeContextual(text);
+					}
+				},
+				
+				_shapeEuropean: function (/*String*/ text){
+					// summary:
+					//		Converts the digits in the text to European digits.
+					// text: String
+					//		The text to be shaped.
+					// return:
+					//		The shaped string in European format.
+					// tags:
+			        //		private
+					
+					var textArr = text.split("");
+					var c = "";
+					for (var i = 0; i < text.length; ++i) {
+						c = textArr[i];
+						if (c >= "\u0660" && c <= "\u0669") { // Arabic digits range
+							textArr[i] = c.charCodeAt(0) - 1632;
+						}
+					}
+					return textArr.join("");
+				},
+				
+				_shapeArabic: function(/*String*/ text){
+					// summary:
+					//		Converts the digits in the text to Arabic digits.
+					// text: String
+					//		The text to be shaped.
+					// return:
+					//		The shaped string in Arabic format.
+					// tags:
+			        //		private
+					
+					var textArr = text.split("");
+					var c = "";
+					for (var i = 0; i < text.length; ++i) {
+						c = textArr[i];
+						if (c >= "0" && c <= "9") { // European digits range
+							textArr[i] = String.fromCharCode(parseInt(c) + 1632);
+						}
+					}
+					return textArr.join("");
+				},
```

Comment:
Looks like `_shapeArabic()` and `_shapeEuropean()` could be implemented more efficiently and with less code using regexs, i.e. something like this for `_shapeArabic()`:

```js
return text.replace(/[0-9]/g, function(c){ return String.fromCharCode(parseInt(c) + 1632); });
```